geoh5py/objects/geo_image.py

```diff
@@ -33,13 +41,204 @@ class GeoImage(ObjectBase):
+                raise ValueError(
+                    "Shape of the 'image' must be a 2D or "
+                    "a 3D array with shape(*,*, 3) representing 'RGB' values."
+                )
+            value = image.astype(float)
+            value -= value.min()
+            value *= 255.0 / value.max()
+            value = value.astype("uint8")
+            image = Image.fromarray(value)
+        elif isinstance(image, str):
+            if not os.path.exists(image):
+                raise ValueError(f"Input image file {image} does not exist.")
+            image = Image.open(image)
+        elif isinstance(image, bytes):
+            image = Image.open(BytesIO(image))
+        elif not isinstance(image, Image.Image):
+            raise ValueError(
+                "Input 'value' for the 'image' property must be "
+                "a 2D or 3D numpy.ndarray, bytes, PIL.Image or a path to an existing image."
+            )
+
+        with TemporaryDirectory() as tempdir:
+            ext = getattr(image, "format")
+            temp_file = os.path.join(
+                tempdir, f"image.{ext.lower() if ext is not None else 'tiff'}"
+            )
+            image.save(temp_file)
+
+            if self.image_data is not None:
+                self.workspace.remove_entity(self.image_data)
+
+            image = self.add_file(temp_file)
+            image.name = "GeoImageMesh_Image"
+            image.entity_type.name = "GeoImageMesh_Image"
+
+        self.vertices = self.default_vertices
+
+    def georeference(self, reference: np.ndarray | list, locations: np.ndarray | list):
+        """
+        Georeference the image vertices (corners) based on input reference and
+        corresponding world coordinates.
+
+        :param reference: Array of integers representing the reference used as reference points.
+        :param locations: Array of floats for the corresponding world coordinates
+            for each input pixel.
+
+        :return vertices: Corners (vertices) in world coordinates.
+        """
+
+        reference = np.asarray(reference)
+        locations = np.asarray(locations)
+        if self.image is None:
+            raise AttributeError("An 'image' must be set before georeferencing.")
+
+        if reference.ndim != 2 or reference.shape[0] < 3 or reference.shape[1] != 2:
+            raise ValueError(
+                "Input reference points must be a 2D array of shape(*, 2) "
+                "with at least 3 control points."
+            )
+
+        if (
+            locations.ndim != 2
+            or reference.shape[0] != locations.shape[0]
+            or locations.shape[1] != 3
+        ):
+            raise ValueError(
+                "Input 'locations' must be a 2D array of shape(*, 3) "
+                "with the same number of rows as the control points."
+            )
+        param_x, _, _, _ = np.linalg.lstsq(
+            np.c_[np.ones(3), reference], locations[:, 0], rcond=None
```

Comment:
should this be np.ones(len(reference))?

Reply:
Very nice catch brother. That's what I am talking about!

Reply:
Fixed